ProfileManager.js

```diff
@@ -422,121 +448,187 @@ export default class ProfileManager {
-    const profileZcapKey = new AsymmetricKey({
-      capability: zcap,
-      invocationSigner,
-      kmsClient
-    });
-
-    return {
-      invocationSigner: profileZcapKey,
-      kmsClient,
-      // FIXME: `zcaps` are coming out of the capabilitySet EDV which is
-      // read deep in the call stack. Need to clean this up.
-      zcaps
-    };
+  // FIXME: expose this or not?
+  async createEdvRecipientKeys({profileId} = {}) {
+    const keystoreAgent = await this.getProfileKeystoreAgent({profileId});
+    const [keyAgreementKey, hmac] = await Promise.all([
+      keystoreAgent.generateKey({
+        type: 'keyAgreement',
+        kmsModule: this.kmsModule,
+      }),
+      keystoreAgent.generateKey({
+        type: 'hmac',
+        kmsModule: this.kmsModule,
+      })
+    ]);
+    return {hmac, keyAgreementKey};
   }
 
-  async updateUser({profileAgent, usersReferenceId, content}) {
-    if(!usersReferenceId) {
-      usersReferenceId = edvs.getReferenceId('users');
+  async getAccessManager({profileId} = {}) {
+    if(typeof profileId !== 'string') {
+      throw new TypeError('"profileId" must be a string.');
     }
-    const {edvClient, invocationSigner} = await this.getUsersEdv(
-      {profileAgent, referenceId: usersReferenceId});
-
-    const [userDoc] = await edvClient.find({
-      equals: {'content.id': content.id},
-      invocationSigner
-    });
-
-    const updatedUserDoc = await edvClient.update({
-      doc: {
-        ...userDoc,
-        content: {
-          ...userDoc.content,
-          ...content
-        }
-      },
-      invocationSigner,
-      keyResolver
+    const [profile, agent] = await Promise.all([
+      this.getProfile({id: profileId}),
+      this.getAgent({profileId})
+    ]);
+    const invocationSigner = await this._getAgentSigner({id: agent.id});
+    // TODO: consider consolidation with `getProfileEdv`
+    const capability = agent.zcaps['user-edv-documents'];
+    const userKak = agent.zcaps['user-edv-kak'];
+    const userHmac = agent.zcaps['user-edv-hmac'];
+    if(!(capability && userKak && userHmac)) {
+      throw new Error(
+        `Profile agent "${agent.id}" is not authorized to manage access ` +
+        `for profile "${profileId}".`);
+    }
+    const {edvId, indexes} = profile.accessManagement;
+    const edvClient = new EdvClient({
+      id: edvId,
+      keyResolver,
+      keyAgreementKey: new KeyAgreementKey({
+        id: userKak.invocationTarget.id,
+        type: userKak.invocationTarget.type,
+        capability: userKak,
+        invocationSigner
+      }),
+      hmac: new Hmac({
+        id: userHmac.invocationTarget.id,
+        type: userHmac.invocationTarget.type,
+        capability: userHmac,
+        invocationSigner
+      })
     });
-    return updatedUserDoc.content;
+    for(const index of indexes) {
+      edvClient.ensureIndex(index);
+    }
+    const users = new Collection(
+      {type: 'User', client: edvClient, capability, invocationSigner});
```

Comment:
possibly just make `client` be `edvClient`.  And maybe Collection should be EdvCollection?

Reply:
I'm fine with changing to `client`. Don't want to change `Collection` right now.

Reply:
I'm suggesting that the param to the Collection constructor be `edvClient` etc.  This is just one instance where `edvClient` is being assigned to `client` in this API.  Is that the change you are amenable to?

Reply:
I'm fine with calling the param `edvClient` inside Collection.